### ISSUE TITLE
Using image from the network

### ISSUE THREAD

**Issue by tawrahim** [Orderella/PopupDialog#3]
Is there a way to use an image from the network? I am thinking perhaps you could somehow expose the reference to `ImageView` so that it can be used after the async call of loading the image is completed.

**Comment by mwfire** [comment]
I will look into this. This makes sense, though the image view height is calculated to resize according to the image dimensions. In case of async loading, the dialog would have to resize after the image was loaded, which might look awkward. An option might be using a placeholder image with the same size, or specifying the image dimensions upfront, which would be less dynamic on the other hand.

For a start I will look into exposing the imageView, as you mentioned.

**Comment by tawrahim** [comment]
Great and thanks! Looking forward to it :+1:

**Comment by mwfire** [comment]
I have exposed all init parameters so that they can be set after initialization as well with 8b5bb46

``` swift
alert.buttonAlignment = .Horizontal
alert.image = UIImage(named: "cute_cat")
alert.titleText = "..."
alert.messageText = "..."
```

Changes can be found on a separate branch, as neither tests have been written nor the readme was updated. I'll see if I can make it this week. The API might change until then.
For now, you should be able to use this version by specifying the branch in the podfile.

`pod 'PopupDialog', :git => 'https://github.com/Orderella/PopupDialog.git', :branch => 'feature/exposeImageView'`

Please let me know what you think @tawrahim

**Comment by tawrahim** [comment]
First of all I appreciate your quick response. I have been playing extensively with your solutions and here are few comments that I have.

Lets assume that the we are using [alamofire](https://github.com/Alamofire/Alamofire) to load our image from the network, we end up doing something along the lines 

``` swift
alert.buttonAlignment = .Horizontal
alert.image = UIImage(named: "cute_cat")
alert.titleText = "..."
alert.messageText = "..."

Alamofire.request(.GET, "https://robohash.org/123.png").response { (request, response, data, error) in
    alert.image = UIImage(data: data!, scale:1)
     self.presentViewController(alert, animated: true, completion: nil)
    }
```

In the completion block is where we present the view controller. The problem I am seeing with this is that, lets say something happened and we were not able to load the image, the popup would never show up. Another thing is that it might take a while for the image to finish loading from the network which may not be the best UX.

On the other hand, I think if we expose a reference to `UIImageView` itself, we wont be forced to `presentViewController` after the image has loaded from the network but rather set the image after the PopupDialog has been present.

Let me know what you think.

**Comment by mwfire** [comment]
The image property I added enables you to do that. I have updated the readme as well, you might have look. The following should work as expected.

``` swift
let alert = PopupDialog(title: "...", message: "...")
// ...
presentViewController(alert, animated: true, completion: nil)

Alamofire.request(.GET, "https://robohash.org/123.png").response { (request, response, data, error) in
    // Image will be updated even though the alert has been displayed
    // This might have to be dispatched to the main thread
    alert.image = UIImage(data: data!, scale:1)
}
```

However, this is something we would not use in our app. We do use this as an alert view with images that come bundled with the app only as, as you mentioned, the UX with async image loading might not be the best.

One issue with this approach is that your request might finish after the dialog has been dismissed already....

**Comment by tawrahim** [comment]
Ah I see! I think you are absolutely right with the user dismissing the dialog even before the image is done loading. Its kind of a chicken-and-egg scenario but overall I appreciate you exposing those properties.

**Comment by mwfire** [comment]
Sure thing! This is now available on the master branch, 19cc279, V 0.1.5
